lib/project_types/script/layers/infrastructure/languages/task_runner.rb

```diff
@@ -15,9 +17,36 @@ def self.for(ctx, language, script_name)
           end
 
           def check_system_dependencies!
-            raise NotImplementedError
+            self.class::REQUIRED_TOOL_VERSIONS.each { |tool| check_tool_version!(tool[:tool_name], tool[:min_version]) }
```

Comment:
I know @ErinRenNumber1 suggested the approach with adding tool versions and I'm not sure what she had in mind, so I'll leave her to nitpick this. But, it's generally bad practice for a parent class to rely on a constant written in a child class. It doesn't create a great interface for child classes to implement because they have to read and fully understand the code to see what constants are required or not. Its also a little wonky because this parent class has code to check the current class of the instance and read a constant, so from reading the code you would expect this parent class to define the constant, but it is nowhere to be found 🤯  

But if you want to go this route by defining constants, you can create a skeleton method in the parent class and expose the constant that way.

```ruby
class Parent
  # ...
  def required_tool_versions
    raise NotImplementedError
  end
end

class Child
  REQUIRED_TOOL_VERSIONS = {}
  # ...
  def required_tool_versions
    REQUIRED_TOOL_VERSIONS
  end
end
```

But before you change anything, I'd recommend chatting with Erin to get her thoughts on how she was thinking this should look!